libc-bottom-half/crt/crt1-command.c

```diff
@@ -7,12 +10,14 @@ __attribute__((export_name("_start")))
 void _start(void) {
     // Commands should only be called once per instance. This simple check
     // ensures that the `_start` function isn't started more than once.
-    static volatile int started = 0;
 #ifdef _REENTRANT
-    if (__sync_val_compare_and_swap(&started, 0, 1) != 0) {
+    static volatile _Atomic int started = 0;
+    int expected = 0;
+    if (!atomic_compare_exchange_strong(&started, &expected, 1)) {
 	__builtin_trap();
     }
 #else
+    static volatile int started = 0;
```

Comment:
No need for volatile here I think?

Reply:
I've now added a comment about `volatile`. It's admittedly a very theoretical concern. But also, this variable is read and written to exactly once, so there's nothing to optimize about it. So I'm inclined to keep the `volatile`.

Reply:
Persisting an in-memory variable past the point of an `_exit` call isn't something that's even possible to think about in C, and yet that's what this code is doing, so `volatile` helps make sure that the compiler just does what we're telling it to do, because it won't understand what we're actually doing.